hibernate-core/src/main/java/org/hibernate/dialect/function/CastFunction.java

```diff
@@ -27,32 +28,52 @@
  */
 public class CastFunction extends AbstractSqmSelfRenderingFunctionDescriptor {
 
-	private Dialect dialect;
+	private final Dialect dialect;
+	private final CastType booleanCastType;
```

Comment:
I would just save the `int preferredSqlTypeCodeForBoolean` instead of the `CastType` here. That way you need one less method.

Reply:
Not sure I understand. Do you want me to move the switch block of `getBooleanCastType` into the if of `getCastType`?

Reply:
Well you have one method which translates an `int` into a `CastType`, which you save in this field.

And then another method which use this `CastType` to do something else.

I would just save away the `int`, not the `CastType` and have just the second method.

Reply:
Is that a request or a comment? I like it this way, but if you insist I'll change it.

Reply:
It's no big deal.